src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotifications.java

```diff
@@ -120,22 +120,35 @@ private static void createStatus(@NonNull Run<?, ?> build, @NonNull TaskListener
         String buildDescription = build.getDescription();
         String statusDescription;
         String state;
+        final String successful_state = "SUCCESSFUL";
+        final String failed_state = "FAILED";
         if (Result.SUCCESS.equals(result)) {
             statusDescription = StringUtils.defaultIfBlank(buildDescription, "This commit looks good.");
-            state = "SUCCESSFUL";
+            state = successful_state;
         } else if (Result.UNSTABLE.equals(result)) {
             statusDescription = StringUtils.defaultIfBlank(buildDescription, "This commit has test failures.");
-            state = "FAILED";
+            final SCMSource s = SCMSource.SourceByItem.findSource(build.getParent());
+            if (!(s instanceof BitbucketSCMSource)) {
+                return;
+            }
```

Comment:
This should probably be at the top of the method.  None of these should run on non-bitbucket sources, right?

Reply:
I have moved it to the top. To be honest I am not sure how this method is called and why checks like this are required but I saw it used elsewhere and reused it.